dora/integration/fuse/src/main/java/alluxio/fuse/FuseSignalHandler.java

```diff
@@ -41,12 +51,14 @@ public void handle(Signal signal) {
     LOG.info("Receive signal name {}, number {}, system exiting",
         signal.getName(), signal.getNumber());
     int number = signal.getNumber();
-    if (number == 15) {
+    // SIGTERM - 15
+    // SIGINT - 2
+    // SIGHUP - 1
+    if (number == 15 || number == 2 || number == 1) {
       try {
-        mFuseUmountable.umount(false);
+        mFuseUmountable.destroy();
       } catch (Throwable t) {
-        LOG.error("unable to umount fuse.", t);
-        return;
+        LOG.error("Unable to umount fuse. exiting anyways...", t);
       }
```

Comment:
@lucyge2022 Please explain the expected result/behavior when the FUSE client receives one of these signals.

According to my testing, sending SIGINT/SIGHUP/SIGTERM to fuse, the fuse process will exit, but the fuse mount point is still there, visible in Linux operating system via "mount" command.
I don't think it is the expected result.


Also, as I said before, SIGHUP should be ignored?
If a developer mount the fuse in foreground, and doing some debugging remotely:
./bin/alluxio-fuse mount -f /mnt/some_fuse_mount

and then the remote session is ended (e.g. ssh session is disconnected, terminal is closed, or shell is killed), SIGHUP will be sent to all running process in this session. So the FUSE process will also receive the SIGHUP.

Reply:
According to my testing, sending SIGINT/SIGHUP/SIGTERM to fuse, the fuse process will exit, but the fuse mount point is still there, visible in Linux operating system via "mount" command.
-> yes that's right, that's coz we implement fuse ops with a java process, initially fuse itself will try to install its own signal handler if no one else specify them, it will continue fuse_main_real and exit that func gracefully. but jvm ALWAYS has its own signal hdlr installed, hence this signal handling is never executed, only way to exit fuse_main_real gracefully is to send 'umount' or 'fusermount', full explanation here in : https://tachyonnexus.atlassian.net/browse/AC-1334
by installing our own signal hdlr here explicitly to override JVM signal hdlrs, the result of lingering mount will still be there (since fuse_main_real is still not gracefully exited ) but we have a chance to clean ourselves gracefully, should umount or fusermount doesnt work. The original signal hdlr executing "umount/fusermount" inside of java process is not the correct way, the unmount procedure should always be umount/fusermount -> if fails sends terminating signals -> if still fails kill -9, that's why i changed the alluxio-fuse unmount shell script

 Also, as I said before, SIGHUP should be ignored?
-> sure i can remove this, reason i added it is coz in libfuse (lib/fuse_signals.c:fuse_set_signal_handlers) it install sighdlrs for SIGHUP.

Reply:
OK